src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.xml

```diff
@@ -40,9 +40,14 @@
             <argument type="service" id="file_locator" />
         </service>
 
+        <service id="routing.loader.service.container" class="Symfony\Component\Routing\Loader\DependencyInjection\ServiceRouterLoaderContainer">
+            <argument type="service" id="service_container" />
+            <argument type="tagged_locator" tag="routing.router_loader" index-by="service_id" />
```

Comment:
Actually, why do we need an index-by attribute at all?
Wouldn't adding a compiler pass allow us to remove the attribute altogether?
I may miss something :)

Reply:
The `index-by` attribute is needed to stay backward compatible with the `MicroKernelTrait` that uses an id for the kernel service in its router loader resource configuration (`kernel::loadRoutes`). That also allow everyone to be able to use service ids and not FQCN which is good IMO. I think we might use `static::class.'::loadRoutes'` here but is it backward compatible?